### DISCUSSION TITLE
Test IsTemporary before DeleteAll

### DISCUSSION THREAD

**Discussion by kine** [microsoft/alguidelines#187] [BC Best Practices]
Could be good practice?

Because parameters, even when defined as Temporary record, when passed by reference and you pass real record, you will work with real record incode, which expect temporary. Same for Pages with IsTemporary = true - when you pass real rec, you will work over real rec, and if you are doing deleteall to clear the "buffer", you will clear real table. 

It is good practice (pattern?) to test that you are really working with temp record.

Same as testing if not temp in Event subscribers of tables (OnBefore/After...)

**Comment by waldo1001** [comment]
Absolutely!

**Reply by waldo1001** [reply]
I rather see this as a guideline though :-)
please feel free to describe it ;-)

**Reply by kine** [reply]
Will try, it is matter of time, you know... ;-)

**Reply by kine** [reply]
PR https://github.com/microsoft/alguidelines/pull/189 created
